Add `font_size` parameter for `Tex` and `Text` mobjects.

hydrobeam opened the pull request:
<!--
Thanks for your contribution to ManimCommunity!

Before filling in the details, ensure:
- Your local changes are up-to-date with ManimCommunity/manim
  
- The title of your PR gives a descriptive summary to end-users. Some examples:
  - Fixed last animations not running to completion
  - Added gradient support and documentation for SVG files
  Examples of what *NOT* to do:
  - "fixed that styling issue" - not descriptive enough
  - "fixed issue #XYZ" - end-user needs to do further research
-->
## Changelog / Overview
<!-- Optional (Recommended): a detailed overview of the PR for the upcoming
release's changelog entry. Useful for when the PR title isn't enough. 

DO NOT REMOVE THE FOLLOWING CHANGELOG LINES, EVEN IF YOU DON'T USE THEM.-->
<!--changelog-start-->

<!--changelog-end-->

## Motivation
<!-- In what way do your changes improve the library? -->
It can feel rather unintuitive to scale text using `.scale()` immediately after creation. A more traditional way of editing the size of text is to specify a font size that is higher than the actual size of the mobject. It also allows for easier fine_tuning of mobject size : `font_size= 25` as opposed to working with decimals with `.scale()`

## Explanation for Changes
<!-- How do your changes improve the library? -->

<!-- For PRs introducing new features, please adjust
the link below to reference the docs of your feature. -->

default of `font_size = 48`. 

NOTE: 
`Tex`/`Text` mobjects all have the same default size:

```py
class mark(Scene):
    def construct(self):
        a = Tex("what").center()
        b = Text(f"{a.get_top() - a.get_bottom()}").scale(0.4).center().shift(UP * 2)
        c = Text(f"{a.get_right() - a.get_left()}").scale(0.4).center().shift(UP)
        self.add(a, b, c)
```
New:
![image](https://user-images.githubusercontent.com/70682032/122000656-2fb61400-cd7d-11eb-8dcb-6df5d3139fb3.png)

Old:
![image](https://user-images.githubusercontent.com/70682032/122000736-507e6980-cd7d-11eb-9792-5b5dff27907d.png)
(they're the same thing)

### `numbers.py`
`numbers.py` refactored to be split into functions.
- `set_value` simplified and more efficient (?)
- removed unneeded code that existed for webgl compatibility

### `coordinate_systems.py`
- All uses of `number_scale_value` removed and replaced with `font_size`
- `PolarPlane` adjusted to accept font_size not scale values

### `numberline.py` 
- Now takes `font_size` instead of `number_scale_value`
- `font_size` parameter added

### `text_mobject.py / tex_mobject.py` 
- the scale factors were adjusted to be 1/960 ( the value they needed to be for the default value to be a 48 and still match the old look exactly)

`Code`, `Paragraph`, `Text`, `Tex`, `MathTex` and `DecimalNumber` support `font_size` now.
 
[Documentation Reference](https://manimce--####.org.readthedocs.build)
<!-- In the link above replace #### with you PR number.
You can also adjust the path to the module / class you worked on. This could look like adding
"/en/####/reference/manim.mobject.geometry.Arc.html" to the link for the Arc class.
Notice that the link will only work once the documentation is build which may take a few minutes.-->


## Testing Status
<!-- Optional (Recommended): your computer specs and what tests you ran with
their results, if any. This section is also intended for other
testing-related comments. -->
Had to update the `NumberPlane` test because it was getting errors on only three of the numbers for some reason.
All other tests pass.

## Further Comments
<!-- Optional: any further comments that might be useful for reviewers. -->
I updated some of the docs so that there would at least be some representation of `font_size`, but I wasn't sure if I should continue updating all places in the docs/code where `scale` is used on text with `font_size` instead.

There were still some leftover classes that used scale values, namely `Brace`. 

Also, I noticed that `Text` is ~1.5x larger than its `Tex` counterparts at least on my machine. Should this be adjusted so that they're both equal?

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings
<!-- Once again, thanks for contributing to ManimCommunity! -->


<!-- Do not modify the lines below. These are for the reviewers of your PR -->
## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

naveen521kk commented on manim/mobject/svg/text_mobject.py:
```diff
@@ -399,6 +399,7 @@ def __init__(
         stroke_width: int = 0,
         color: str = WHITE,
         size: int = 1,
+        font_size=48,
```
Users would be confused by having both `size` and `font_size` parameters.

hydrobeam replied:
> I would like it to be called scale_factor.

Even though this parameter is effectively just scaling, I think the idea of a `font_size` is something that most people can easily grasp and work with, especially when dealing with text. Renaming the parameter to `scale_factor` would be less intuitive. In my opinion, how it really works under the hood isn't the biggest concern, because if the user can see that it works like a `font_size` and it meets their expectations, then it's acceptable.

> Also, Pango can handle Font Size (the size parameter in Text)

We can have `Text` use the appropriate method of handling Font Size when a fix is implemented.

> Users would be confused by having both size and font_size parameters.

Since `size` isn't working at the moment, I don't think there are many people that are using it to adjust their text. We can probably add some documentation to explain the differences between the two.

naveen521kk replied:
> Even though this parameter is effectively just scaling, I think the idea of a font_size is something that most people can easily grasp and work with, especially when dealing with text. Renaming the parameter to scale_factor would be less intuitive. In my opinion, how it really works under the hood isn't the biggest concern, because if the user can see that it works like a font_size and it meets their expectations, then it's acceptable.

Changing font size is never the same as scaling. There are ways to change font size Tex also but that isn't implemented https://texblog.org/2012/08/29/changing-the-font-size-in-latex/ and for `Text`, it is already there as `size` parameter(maybe you can rename them to `font_size` if you want to). 

> We can have Text use the appropriate method of handling Font Size when a fix is implemented.

It isn't broken entirely. For example, this works
```py
def construct(self):
  self.texts = [Text(f'hello{i}',size=i) for i in range(1,30)]
  for i in self.texts:
    self.add(i.scale(0.4))
    self.wait()
    self.clear()
```

https://user-images.githubusercontent.com/49693820/122533767-8a7e9280-d03f-11eb-8e8c-21c111ce0465.mp4

And I don't want to confuse users by introducing another parameter.

If the end goal is to increase the font size there are ways to do that currently rather than scaling.

hydrobeam replied:
It's reasonable to not want to introduce a new parameter. 

I noticed [this commit](https://github.com/3b1b/manim/commit/6eb7edc664d48498fa53aefabf838c2aa80782fa) by you from the manimgl branch regarding text scaling. Would you instead prefer deprecating `size` and using the built in scaling as opposed to scaling? 

I can also look into the Tex font size, but from what I've seen it's generally preferred to not use strict numbers for font_size and instead something like `\huge`, `\small` etc.. It is possible to use `\fontsize` as described [here](https://www.sascha-frank.com/latex-font-size.html) but it can also cause errors when a font is not supported. Also, scaling in math mode seems to be less straightforward and seems like it entails scaling [from stackexchange](https://tex.stackexchange.com/questions/60453/reducing-font-size-in-equation). So, I think having `font_size` simply scale text is fine for `MathTex`/`Tex`

naveen521kk replied:
> Would you instead prefer deprecating size and using the built in scaling as opposed to scaling?

Fine with me.

> I can also look into the Tex font size, but from what I've seen it's generally preferred to not use strict numbers for font_size and instead something like `\huge`, `\small` etc.. It is possible to use `\fontsize` as described [here](https://www.sascha-frank.com/latex-font-size.html) but it can also cause errors when a font is not supported. Also, scaling in math mode seems to be less straightforward and seems like it entails scaling: [from stackexchange](https://tex.stackexchange.com/questions/60453/reducing-font-size-in-equation). So, I think having `font_size` simply scale text is fine for `MathTex`/`Tex`

Not sure about this. what do you this about this @ManimCommunity/latex?

behackl replied:
Scaling is fine for now. It usually isn't a big problem to use `\fontsize{x}{y}\selectfont`, but this PR shouldn't be delayed over it. We can investigate / think about it later.

naveen521kk reviewed (commented):
I don't like calling the parameter `font_size`, as it just scales the Text rather than actually increasing the font size. Scaling isn't the same as setting font size, a Text rendering engine (say it is LaTeX or Pango) would have to other things while they increase the font size. I would like it to be called `scale_factor`.

Also, Pango can handle Font Size (the `size` parameter in `Text`), but it doesn't work because of the SVG renderer https://github.com/ManimCommunity/manim/issues/1390. I have no idea about LaTeX though.

hydrobeam commented:
`text_mobject.py` now uses pango's font_size scaling. 

I've also added a `DEFAULT_FONT_SIZE` parameter, updated to using  `super()` in some places, added font_size scaling for `Brace` and `Title`. Also, I've adjusted the parameters in `text2svg`  so that and the size of the mobject remains the same as before.

naveen521kk commented on manim/mobject/svg/text_mobject.py:
```diff
@@ -56,6 +56,7 @@ def construct(self):
 
 import manimpango
 import numpy as np
+from colour import Color
```
NIT: you have imported this just for adding typings so this should be under the check for https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

behackl commented on tests/test_texmobject.py:
```diff
@@ -72,3 +72,12 @@ def test_tex_white_space_and_non_whitespace_args():
+    text = Tex("what").center()
+    vertical = text.get_top() - text.get_bottom()
+    horizontal = text.get_right() - text.get_left()
+    assert vertical[1] == 0.35118305000000005
+    assert horizontal[0] == 1.0419676499999995
```
Tests with this sort of accuracy are a recipe for misery. I don't think that we should test more than the first 3 or 4 decimal places.

behackl reviewed (commented):
The idea here seems fine to me, although in general @naveen521kk is of course right: font size != scaling in general.

However, this refactor does clean up the scaling interface in general by a fair amount, and if we decide to make this work cleanly at some later point, it would be easier to do so. I'd support going through with this.

It would be good if `Text` and `Tex` were similarly large if the same font size parameter is passed to them.

PhotonSpheres commented:
> It would be good if `Text` and `Tex` were similarly large if the same font size parameter is passed to them.

I second this. Also that in general LaTeX or Pango should handle something called `font_size` internally rather than a manim `scale`. But this might be over the scope of this PR.

hydrobeam commented:
I've resolved the conflicts and also deprecated `number_scale_value` and updated some examples. For some reason adding `if TYPE_CHECKING` raised some errors, so I've just dropped that for now.


> It would be good if Text and Tex were similarly large if the same font size parameter is passed to them.

I agree, it's a little odd that they vary. We might have some problems with that since the size of `Text` varies per OS.

hydrobeam commented:
- Rewrote some tests: (unnoticeable visual differences)
- Added a `font_size` property so that size can be adjusted afterwards.
- Values passed to `add_labels` now must have a `font_size` attribute since it's used to set the size of the label.

Notes:
- the font_size property for `text_mobject.py` uses scaling to scale. Eventually we can try to make it use natural font size scaling.
- sometimes strings are generated with `self.height` equal 0, hence the check for height in `svg_tex_mobject.py`: [example to see this effect with a debugger](https://docs.manim.community/en/stable/reference/manim.animation.indication.ShowPassingFlash.html). `SingleStringMathTex("}")` with 0 height.


Side notes:
- `text_mobject.py` has a *lot* of needless code duplication, and is hard to understand to understand since it's scarcely documented 🙁,

Darylgolden commented on docs/source/tutorials/using_text.rst:
```diff
@@ -469,5 +470,5 @@ when typesetting multiline formulae:
     class LaTeXAlignEnvironment(Scene):
         def construct(self):
-            tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6').scale(2)
+            tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6',font_size=96)
```
```suggestion
            tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6', font_size=96)
```

Darylgolden commented on manim/animation/creation.py:
```diff
@@ -15,17 +15,17 @@ def construct(self):
-            t6 = Text("ShowIncreasingSubsets").scale(0.45).next_to(s6, UP)
-            t7 = Text("ShowSubmobjectsOneByOne").scale(0.45).next_to(s7, UP)
+            t1 = Text("Write", font_size=24).next_to(s1, UP)
+            t2 = Text("AddTextLetterByLetter", font_size = 24).next_to(s2, UP)
```
```suggestion
            t2 = Text("AddTextLetterByLetter", font_size=24).next_to(s2, UP)
```

Darylgolden commented on manim/animation/creation.py:
```diff
@@ -15,17 +15,17 @@ def construct(self):
-            t7 = Text("ShowSubmobjectsOneByOne").scale(0.45).next_to(s7, UP)
+            t1 = Text("Write", font_size=24).next_to(s1, UP)
+            t2 = Text("AddTextLetterByLetter", font_size = 24).next_to(s2, UP)
+            t3 = Text("Create", font_size = 24).next_to(s3, UP)
```
```suggestion
            t3 = Text("Create", font_size=24).next_to(s3, UP)
```

Darylgolden commented on manim/animation/creation.py:
```diff
@@ -15,17 +15,17 @@ def construct(self):
+            t1 = Text("Write", font_size=24).next_to(s1, UP)
+            t2 = Text("AddTextLetterByLetter", font_size = 24).next_to(s2, UP)
+            t3 = Text("Create", font_size = 24).next_to(s3, UP)
+            t4 = Text("Uncreate", font_size = 24).next_to(s4, UP)
```
```suggestion
            t4 = Text("Uncreate", font_size=24).next_to(s4, UP)
```

Darylgolden commented on manim/animation/creation.py:
```diff
@@ -15,17 +15,17 @@ def construct(self):
+            t2 = Text("AddTextLetterByLetter", font_size = 24).next_to(s2, UP)
+            t3 = Text("Create", font_size = 24).next_to(s3, UP)
+            t4 = Text("Uncreate", font_size = 24).next_to(s4, UP)
+            t5 = Text("DrawBorderThenFill", font_size = 24).next_to(s5, UP)
```
```suggestion
            t5 = Text("DrawBorderThenFill", font_size=24).next_to(s5, UP)
```

Darylgolden commented on manim/animation/creation.py:
```diff
@@ -15,17 +15,17 @@ def construct(self):
+            t3 = Text("Create", font_size = 24).next_to(s3, UP)
+            t4 = Text("Uncreate", font_size = 24).next_to(s4, UP)
+            t5 = Text("DrawBorderThenFill", font_size = 24).next_to(s5, UP)
+            t6 = Text("ShowIncreasingSubsets", font_size = 22).next_to(s6, UP)
```
```suggestion
            t6 = Text("ShowIncreasingSubsets", font_size=22).next_to(s6, UP)
```

Darylgolden commented on manim/animation/creation.py:
```diff
@@ -15,17 +15,17 @@ def construct(self):
+            t4 = Text("Uncreate", font_size = 24).next_to(s4, UP)
+            t5 = Text("DrawBorderThenFill", font_size = 24).next_to(s5, UP)
+            t6 = Text("ShowIncreasingSubsets", font_size = 22).next_to(s6, UP)
+            t7 = Text("ShowSubmobjectsOneByOne", font_size = 22).next_to(s7, UP)
```
```suggestion
            t7 = Text("ShowSubmobjectsOneByOne", font_size=22).next_to(s7, UP)
```

Darylgolden commented on manim/animation/creation.py:
```diff
@@ -15,17 +15,17 @@ def construct(self):
             self.add(s1, s2, s3, s4, s5, s6, s7, t1, t2, t3, t4, t5, t6, t7)
 
-            texts = [Text("manim").scale(0.6), Text("manim").scale(0.6)]
+            texts = [Text("manim", font_size = 29), Text("manim", font_size = 29)]
```
```suggestion
            texts = [Text("manim", font_size=29), Text("manim", font_size=29)]
```

Darylgolden commented on manim/mobject/geometry.py:
```diff
@@ -435,7 +435,7 @@ def __init__(self, start, end, angle=TAU / 4, radius=None, **kwargs):
             angle = math.acos((radius - arc_height) / radius) * sign
 
-        Arc.__init__(self, radius=radius, angle=angle, **kwargs)
+        super().__init__(radius=radius, angle=angle, **kwargs)
```
Is this relevant for this PR?

hydrobeam replied:
Not really, just caught it while something was breaking.

Darylgolden replied:
I guess we could leave it in.

Darylgolden commented on manim/mobject/svg/code_mobject.py:
```diff
@@ -94,8 +95,8 @@ def construct(self):
-    scale_factor : class:`float`, optional
-        A number which scales displayed code. Defaults to 0.5.
+    font_size : class:`float`, optional
+        A number which scales displayed code. Defaults to 24.
```
```suggestion
        A number which scales displayed code. Defaults to 24. To convert old scale factors to font size, multiply by 48.
```

hydrobeam replied:
It's a little akward to put this in the docs.

Darylgolden replied:
Sure.

Darylgolden commented on manim/mobject/svg/code_mobject.py:
```diff
@@ -148,13 +149,19 @@ def construct(self):
+        params="scale_factor",
+        since="v0.10.0",
+        until="v0.11.0",
+        message="Use font_size instead.",
```
```suggestion
        message="Use font_size instead. To convert old scale factors to font size, multiply by 48.",
```

Darylgolden commented on manim/mobject/svg/tex_mobject.py:
```diff
@@ -486,21 +513,28 @@ def construct(self):
+        params="scale_factor",
+        since="v0.10.0",
+        until="v0.11.0",
+        message="Use font_size instead.",
```
```suggestion
        message="Use font_size instead. To convert old scale factors to font size, multiply by 48.",
```

Darylgolden commented on manim/mobject/three_dimensions.py:
```diff
@@ -76,8 +76,8 @@ def construct(self):
         params="u_min,u_max,v_min,v_max",
-        since="v0.9.0",
-        until="v0.10.0",
+        since="v0.10.0",
```
Why is this changed?

hydrobeam replied:
whoops!

Darylgolden reviewed (changes requested):
Suggested some changes, hope this gets merged soon!

Darylgolden commented on manim/animation/animation.py:
```diff
@@ -80,9 +80,9 @@ def construct(self):
 
                 # Label groups
-                self.add(Text("lag_ratio = ").scale(0.7).next_to(groups, UP, buff=1.5))
+                self.add(Text("lag_ratio = ", font_size = 36).next_to(groups, UP, buff=1.5))
```
```suggestion
                self.add(Text("lag_ratio = ", font_size=36).next_to(groups, UP, buff=1.5))
```

Darylgolden commented on manim/animation/animation.py:
```diff
@@ -80,9 +80,9 @@ def construct(self):
+                self.add(Text("lag_ratio = ", font_size = 36).next_to(groups, UP, buff=1.5))
                 for group, ratio in zip(groups, ratios):
-                    self.add(Text(str(ratio)).scale(0.7).next_to(group, UP))
+                    self.add(Text(str(ratio), font_size = 36).next_to(group, UP))
```
```suggestion
                    self.add(Text(str(ratio), font_size=36).next_to(group, UP))
```

Darylgolden commented on manim/mobject/number_line.py:
```diff
@@ -403,7 +440,7 @@ def __init__(
         numbers_with_elongated_ticks=None,
         decimal_number_config=None,
-        **kwargs
+        **kwargs,
```
```suggestion
        **kwargs
```

Darylgolden reviewed (changes requested):
More minor things

Darylgolden reviewed (approved):
LGTM

hydrobeam commented:
I was doing some quick testing, and noticed that adding the font_size property for text_mobject.py messed with the line spacing, I'll investigate this issue 🤔

behackl commented on docs/source/examples.rst:
```diff
@@ -457,8 +457,8 @@ Special Camera Settings
             image.height = 7
-            frame_text = Text("Frame", color=PURPLE).scale(1.4)
-            zoomed_camera_text = Text("Zoomed camera", color=RED).scale(1.4)
+            frame_text = Text("Frame", color=PURPLE, font_size = 67)
+            zoomed_camera_text = Text("Zoomed camera", color=RED, font_size = 67)
```
```suggestion
            frame_text = Text("Frame", color=PURPLE, font_size=67)
            zoomed_camera_text = Text("Zoomed camera", color=RED, font_size=67)
```

behackl commented on docs/source/examples.rst:
```diff
@@ -627,8 +627,7 @@ Advanced Projects
             grid = NumberPlane()
-            grid_title = Tex("This is a grid")
-            grid_title.scale(1.5)
+            grid_title = Tex("This is a grid", font_size = 72)
```
```suggestion
            grid_title = Tex("This is a grid", font_size=72)
```

behackl commented on manim/mobject/coordinate_systems.py:
```diff
@@ -2103,30 +2105,29 @@ def add_coordinates(
         self.add(self.get_coordinate_labels(r_values, a_values))
         return self
 
-    def get_radian_label(self, number, stacked=True):
```
Was `stacked` actually unused before?

hydrobeam replied:
Yup, it only showed up in the definition of the method. Maybe @ralphieraccoon knows what it was supposed to be?

ralphieraccoon replied:
I think it was originally supposed to be an argument that's now handled by `self.azimuth_compact_fraction` on creation. It looks like an orphan argument that was missed when it was merged, so feel free to delete it.

behackl commented on manim/mobject/number_line.py:
```diff
@@ -30,46 +32,46 @@ class NumberLine(Line):
         Determines whether numbers are added to tick marks. The number of decimal places is determined
         by the step size, this default can be overridden by ``decimal_number_config``.
-    label_direction : Union[:class:`list`, :class:`numpy.ndarray`]
+    font size
```
```suggestion
    font_size
```

behackl replied:
The previous parameter, `number_scale_value` explicitly referenced numbers -- would it make sense to call this parameter `number_font_size`?

hydrobeam replied:
I have a preference for just `font_size`. With the old name, it wouldn't be clear that it was the labels that were being scaled with just `scale_value`, whereas  font size is pretty synonymous with text.

behackl commented on manim/mobject/svg/brace.py:
```diff
@@ -134,17 +135,28 @@ def get_direction(self):
         brace_direction=DOWN,
         label_constructor=MathTex,
-        label_scale=1,
+        font_size=DEFAULT_FONT_SIZE,
```
Similar to above: `font_scale` vs. `label_font_scale`?

hydrobeam replied:
Same reasoning as above

behackl commented on manim/mobject/svg/tex_mobject.py:
```diff
@@ -101,14 +106,38 @@ def __init__(
+        if font_val <= 0:
+            logger.warning("font_size must be greater than 0.")
+        elif self.height > 0:
+            # sometimes manim generates a SingleStringMathex mobject with 0 height. can't be scaled regardless and will error without the elif.
+            # scale to a factor of the initial height so that setting font_size does not depend on current size.
```
These liens are a bit long and should be broken.

behackl reviewed (approved):
Given that others have already reviewed this more carefully, I've (just) read the changes here; they look good to me in general. I have minor formatting suggestions and some comments regarding the naming of some keywords.

The merge conflicts also have to be resolved before this can be merged.

jaysonpowers commented:
I'm getting the error:

`TypeError: __init__() got an unexpected keyword argument 'font_size'`

behackl commented:
> I'm getting the error:
> 
> `TypeError: __init__() got an unexpected keyword argument 'font_size'`

Could you please add an example so that we can reproduce the issue?

jaysonpowers commented:
I'm running the current basic.py scene 1, from the manim example. I just updated my manim to v0.10.0, but it's still running the v0.9.0 that may be the issue.

jaysonpowers commented:
I'm confused, it said to update manim, then it said it was already updated at 0.9.0

behackl commented:
> I'm running the current basic.py scene 1, from the manim example. I just updated my manim to v0.10.0, but it's still running the v0.9.0 that may be the issue.

Don't fully trust the version that is printed when running `manim`, we've had problems with that before. What does `pip list` (or `pip3 list`) say about the installed version?

I can't reproduce the issue with the scenes in `basic.py` – so I'm not sure yet where the problem might lie. Anyhow, this PR is the wrong place to discuss this; would you mind either opening a new issue, or alternatively [send a message in our Discord server](https://www.manim.community/discord/), should you have Discord.

jaysonpowers commented:
Okay
`manimce             0.1.1.post2`
is what I get, is this the correct one? I've uninstalled and reinstalled via pip, I even restarted my computer.

behackl commented:
> Okay
> `manimce 0.1.1.post2`
> is what I get, is this the correct one? I've uninstalled and reinstalled via pip, I even restarted my computer.

Ah! No, you have installed the wrong version: uninstall `manimce`, and install `manim`. Sorry for the confusion with the versions.

jaysonpowers commented:
still geting the error

behackl commented:
> still geting the error

Please open a new issue, include the code you are rendering and the full traceback.

MiltonCobo commented:
I have upgrade to version 0.10.0, and i have checked it using pip list.
When I run any example of MathTex or Tex using font-size I get the error:

__init__() got an unexpected keyword argument 'font_size'

What's going on?

behackl commented:
> I have upgrade to version 0.10.0, and i have checked it using pip list.
> When I run any example of MathTex or Tex using font-size I get the error:
> 
> **init**() got an unexpected keyword argument 'font_size'
> 
> What's going on?

Please open a new issue or discussion and post the full traceback.
